src/main/java/com/habicus/core/model/Goals.java

```diff
@@ -0,0 +1,184 @@
+/*
+ _   _       _     _
+| | | | __ _| |__ (_) ___ _   _ ___
+| |_| |/ _` | '_ \| |/ __| | | / __|
+|  _  | (_| | |_) | | (__| |_| \__ \
+|_| |_|\__,_|_.__/|_|\___|\__,_|___/
+
+ * This file is part of the Habicus Core Platform (https://github.com/Habicus/Habicus-Core).
+ * Copyright (c) 2018 Habicus Core
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.habicus.core.model;
+
+import java.sql.Timestamp;
+import java.util.Objects;
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+
+@Entity
+@IdClass(GoalsPK.class)
+public class Goals {
+
+  private int goalId;
+  private String title;
+  private String description;
+  private String goalInterval;
+  private int usersUserId;
+  private int taskAmount;
+  private Timestamp dueDate;
+  private String labelColor;
+  private Double pledgeAmount;
+  private String goalComplete;
+
+  @Id
+  @Column(name = "goal_id", unique = true)
+  public int getGoalId() {
+    return goalId;
+  }
+
+  public void setGoalId(int goalId) {
+    this.goalId = goalId;
+  }
+
+  @Basic
+  @Column(name = "title")
+  public String getTitle() {
+    return title;
+  }
+
+  public void setTitle(String title) {
+    this.title = title;
+  }
+
+  @Basic
+  @Column(name = "description")
+  public String getDescription() {
+    return description;
+  }
+
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  @Basic
+  @Column(name = "goal_interval")
+  public String getGoalInterval() {
+    return goalInterval;
+  }
+
+  public void setGoalInterval(String goalInterval) {
+    this.goalInterval = goalInterval;
+  }
+
+  @Id
+  @Column(name = "users_user_id")
+  public int getUsersUserId() {
+    return usersUserId;
+  }
+
+  public void setUsersUserId(int usersUserId) {
+    this.usersUserId = usersUserId;
+  }
+
+  @Basic
+  @Column(name = "task_amount")
+  public int getTaskAmount() {
```

Comment:
What is task amount? Is this monetary? We'll probably want a separate class for handling the monetary amount tied to a task.. nonetheless, if this is monetary and we want just a property for now, would a floating point be better suited?

Reply:
Task amount is not monetary, it is the unit associated with the unit number.. I'm willing to discuss naming for the unit number.

Reply:
✔️ Changed the name